src/components/organisms/TestConfigurationForm/Git/SourceEditFormFields.tsx

```diff
@@ -1,29 +1,19 @@
 import {useState} from 'react';
 
-import {NamePath} from 'antd/lib/form/interface';
-
 import {Path, Repository, Revision, SecretFormItem} from '@molecules';
 
-import useValidateRepository from '@hooks/useValidateRepository';
+import useValidateRepository, {ValidationState} from '@hooks/useValidateRepository';
 
 import {useValidateRepositoryMutation} from '@services/repository';
 
 import {StyledFormSpace} from '../TestConfigurationForm.styled';
+import {Props} from '../utils';
 
-type SourceEditProps = {
-  executorType: string;
-  isClearedToken?: boolean;
-  setIsClearedToken?: (value: boolean) => void;
-  isClearedUsername?: boolean;
-  setIsClearedUsername?: (value: boolean) => void;
-  getFieldValue: (name: NamePath) => string;
-};
-
-const SourceEdit: React.FC<SourceEditProps> = props => {
+const SourceEdit: React.FC<Partial<Props>> = props => {
```

Comment:
why do we use Partial here and cast the props to Props afterwards? if not all props are required, we should make the optional in the props interface

Reply:
+1, maybe you may use `Pick<Props, ...>` or `Omit<Props, ...>` to select which are supported